src/hipscat/io/file_io/file_pointer.py

```diff
@@ -33,3 +34,58 @@ def does_file_or_directory_exist(pointer: FilePointer) -> bool:
         True if file or directory at `pointer` exists, False if not
     """
     return os.path.exists(pointer)
+
+
+def is_regular_file(pointer: FilePointer) -> bool:
+    """Checks if a file or directory exists for a given file pointer
```

Comment:
Does this check for directories as well? I think there's already a function for that unless this does something different.

Reply:
Just regular files. Updated comment to match intent.